### PR TITLE
Updating openshift/api, openshift/client-go, and openshift/library-go

### PR DESCRIPTION
**Pull request by kwoodson**
Updating the go.mod and corresponding vendor directories to match the latest openshift/api

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,25 +7,31 @@ require (
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc
 	github.com/ghodss/yaml v1.0.1-0.20190212211648-25d852aebe32 // indirect
 	github.com/go-bindata/go-bindata v3.1.2+incompatible
-	github.com/openshift/api v0.0.0-20210817132244-67c28690af52
+	github.com/go-logr/logr v1.1.0 // indirect
+	github.com/google/go-cmp v0.5.6 // indirect
+	github.com/google/gofuzz v1.2.0 // indirect
+	github.com/openshift/api v0.0.0-20210907212721-c9d6fcc62893
```
Comment (soltysh) [inline go.mod:13]: I'd expect only openshift/api being bumped here. Same comment as with https://github.com/openshift/client-go/pull/194#discussion_r709075079 Also I'd bump client-go as well, when touching it.

File: go.mod [hunk 2]
```diff
@@ -7,25 +7,31 @@ require (
...
 	k8s.io/client-go v0.22.0
 	k8s.io/component-base v0.22.0
-	k8s.io/klog/v2 v2.9.0
+	k8s.io/klog/v2 v2.10.0
 	k8s.io/kube-openapi v0.0.0-20210817084001-7fbd8d59e5b8 // indirect
 	sigs.k8s.io/apiserver-network-proxy/konnectivity-client v0.0.22 // indirect
 	sigs.k8s.io/yaml v1.2.0
 )
+
+replace github.com/go-logr/logr => github.com/go-logr/logr v0.4.0
```
Comment (soltysh) [inline go.mod:37]: That's unexpected as well.

**Review by soltysh — CHANGES_REQUESTED** [review]

**Comment by kwoodson** [comment]
@soltysh Thanks, I'll update this today to include api and client-go only.

File: go.mod (hunk 2 above)
Comment (kwoodson) [reply go.mod]: I have updated this PR to update the openshift libraries.

**Comment by kwoodson** [comment]
@soltysh I have updated the openshift library-go, client-go, and api. 

PTAL

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**